kratos/containers/pointer_vector_set.h

```diff
@@ -653,6 +653,51 @@ class PointerVectorSet final
         SortedInsert(first, new_last);
     }
 
+    /**
+     * @brief Insert elements from a range of iterators from an unordered set.
+     * @details This method allows adding elements from an unordered set, therefore skips the step which
+     *          the normal insert is made to unique since unordered sets are always having unique elements.
+     * @tparam TArgs
+     * @param first An input iterator pointing to the beginning of the range to insert.
+     * @param last An input iterator pointing to the end of the range to insert.
+     */
+    template <class... TArgs>
+    void insert(
+        typename std::unordered_set<TArgs...>::const_iterator first,
+        typename std::unordered_set<TArgs...>::const_iterator last)
+    {
+        using unordered_set_key_type = typename std::unordered_set<TArgs...>::key_type;
+        using unordered_set_key_equal = typename std::unordered_set<TArgs...>::key_equal;
+
+        if constexpr(
+                (
+                    std::is_same_v<unordered_set_key_type, key_type> &&
+                    std::is_same_v<unordered_set_key_equal, std::equal_to<std::remove_cv_t<key_type>>>
+                    // unordered set has used the same key type, and the std::equal_to to add elements.
+                    // hence the given range [first, last] was created by having unique elements while doing the same comparison
+                    // as the PointerVectorSet. Hence, no std::unique is required.
+                )
+                ||
+                (
+                    std::is_pointer_v<unordered_set_key_type> &&
+                    std::is_same_v<unordered_set_key_equal, PointerComparor<std::remove_cv_t<key_type>>>
+                    // unordered set has used the pointer type of the key, and PointerComparor to add elements.
+                    // hence the given range [first, last] was created by having unique elements while doing the same comparison
+                    // as the PointerVectorSet. Hence, no std::unique is required.
+                ))  {
+
+                TContainerType aux_container;
+                aux_container.resize(std::distance(first, last));
```

Comment:
this might be slower than just inserting into a zero capacity vector because finding the size of an unordered set is an O(n) operation.

Reply:
Agree with @matekelemen.

Provided that `PointerVectorSet` is never used with a `TContainerType` that does not have a range constructor (and for that matter, is only used with `std::vector`, so probably we should think about removing that template argument), we should be able to directly use:

```c++
TContainerType aux_container(first, last);

std::sort(aux_container.begin(), aux_container.end(), CompareKey());
SortedInsert(aux_container.begin(), aux_container.end());
```

Which is not a super big performance gain, but might be noticeable in super large sets.